contracts/root/predicates/ERC20Predicate.sol

```diff
@@ -0,0 +1,283 @@
+pragma solidity ^0.5.2;
+
+import { BytesLib } from "../../common/lib/BytesLib.sol";
+import { Common } from "../../common/lib/Common.sol";
+import { Math } from "openzeppelin-solidity/contracts/math/Math.sol";
+import { RLPEncode } from "../../common/lib/RLPEncode.sol";
+import { RLPReader } from "solidity-rlp/contracts/RLPReader.sol";
+
+import { IErcPredicate } from "./IPredicate.sol";
+import { WithdrawManagerHeader } from "../withdrawManager/WithdrawManagerStorage.sol";
+
+contract ERC20Predicate is IErcPredicate {
+  using RLPReader for bytes;
+  using RLPReader for RLPReader.RLPItem;
+
+  bytes32 constant DEPOSIT_EVENT_SIG = 0x4e2ca0515ed1aef1395f66b5303bb5d6f1bf9d61a353fa53f73f8ac9973fa9f6;
+  bytes32 constant WITHDRAW_EVENT_SIG = 0xebff2602b3f468259e1e99f613fed6691f3a6526effe6ef3e768ba7ae7a36c4f;
+  bytes32 constant LOG_TRANSFER_EVENT_SIG = 0xe6497e3ee548a3372136af2fcb0696db31fc6cf20260707645068bd3fe97f3c4;
+  // 0x2e1a7d4d = keccak256('withdraw(uint256)').slice(0, 4)
+  bytes4 constant WITHDRAW_FUNC_SIG = 0x2e1a7d4d;
+  // 0xa9059cbb = keccak256('transfer(address,uint256)').slice(0, 4)
+  bytes4 constant TRANSFER_FUNC_SIG = 0xa9059cbb;
+  bytes constant public networkId = "\x0d";
```

Comment:
Should we use some function for networkId? Changing contract for deployments is painful!

Reply:
no, `networkId` isn't supposed to change once the contracts deployed.

Reply:
We will have the `networkId` decided before deploying the contracts. What's incorrect here is that networkId is duplicated at several places. I'll maintain just one value in `withdrawManager` and pass that value around.

Reply:
@atvanguard yes. At, one place and constant. No need to make it changeable.

Reply:
Done.